src/MatrixFields/field_matrix_iterative_solver.jl

```diff
@@ -411,23 +415,29 @@ check_field_matrix_solver(alg::StationaryIterativeSolve, cache, A, b) =
 
 function run_field_matrix_solver!(alg::StationaryIterativeSolve, cache, x, A, b)
     P = lazy_or_concrete_preconditioner(alg.P_alg, cache.P_cache, A)
-    using_cuda = ClimaComms.array_type(concrete_field_vector(b)) <: CUDA.CuArray
-    !using_cuda && @debug begin
-        e₀ = concrete_field_vector(b) # Initialize e to any nonzero vector.
-        λs, _, info = KrylovKit.eigsolve(e₀, 1; alg.eigsolve_kwargs...) do e
-            e_view = field_vector_view(e, keys(b).name_tree)
-            lazy_Ae = lazy_mul(A, e_view)
-            lazy_Δe = apply_preconditioner(alg.P_alg, cache.P_cache, P, lazy_Ae)
-            concrete_field_vector(@. e_view - lazy_Δe) # (I - inv(P) * A) * e
-        end
-        if info.converged == 0
-            (; tol, maxiter) = alg.eigsolve_kwargs
-            "Unable to approximate ρ(I - inv(P) * A) to within a tolerance \
-            of $(100 * tol) % in $maxiter or fewer iterations"
-        else
-            "ρ(I - inv(P) * A) ≈ $(abs(λs[1]))"
-        end
-    end _group = :spectral_radius
+    if get_debug(alg)
+        @debug begin
+            e₀ = concrete_field_vector(b) # Initialize e to any nonzero vector.
+            λs, _, info = KrylovKit.eigsolve(e₀, 1; alg.eigsolve_kwargs...) do e
+                e_view = field_vector_view(e, keys(b).name_tree)
+                lazy_Ae = lazy_mul(A, e_view)
+                lazy_Δe = apply_preconditioner(
+                    alg.P_alg,
+                    cache.P_cache,
+                    P,
+                    lazy_Ae,
+                )
+                concrete_field_vector(@. e_view - lazy_Δe) # (I - inv(P) * A) * e
+            end
+            if info.converged == 0
+                (; tol, maxiter) = alg.eigsolve_kwargs
+                "Unable to approximate ρ(I - inv(P) * A) to within a tolerance \
+                of $(100 * tol) % in $maxiter or fewer iterations"
+            else
+                "ρ(I - inv(P) * A) ≈ $(abs(λs[1]))"
+            end
+        end _group = :spectral_radius
```

Comment:
```suggestion
        end
```
If we're going to toggle debugging at the `struct` level, let's remove toggling at the logger level. It'll be a bit of a hassle to remember to toggle both the struct flag and the value of `ENV["JULIA_DEBUG"]`. Can we get rid of the `_group` restrictions and change the `@debug`s to `@info`s?

Reply:
Alternatively, can we set the default value of `debug` based on whether `ENV["JULIA_DEBUG"]` contains the substrings `"error_norm"` or `"spectral_radius"`? That's probably a bit more generalizable.

Reply:
Sounds good to me. I'll make `debug` depend on `ENV["JULIA_DEBUG"]`